src/core/tsi/gts_transport_security.c

```diff
@@ -0,0 +1,35 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/tsi/gts_transport_security.h"
+#include <grpc/support/alloc.h>
+
+static gts_shared_resource* gts_resource = NULL;
```

Comment:
Who's going to set this pointer, and when will that happen?

If we really want a single global object here, why not just have this be the struct itself, instead of a pointer that has to be set later?  The initialization hook can initialize the data members.